### ISSUE TITLE
Qualweb/core and online evaluator cannot evaluate certain sites

### ISSUE THREAD

**Issue by Tellus** [qualweb/core#12]
Hey guys,

We've hit some broad issues on two sites that we're trying to run through Qualweb. I've written down details on both sites in this issue, I hope that's acceptable. I can happily report that we've run through several dozen others without any major issues, though! :-)

Given how broad the errors seem to be (essentially the entirety of both sites), I suspect that the problem might lie in a QW implementation rather than the case for single misbehaving pages. I'm hoping you can help us sort this out.

Site 1
------

Here's a handful of URLs for the first site. The URLs fail in both our production and development environments (qualweb-core versions 0.4.33 and 0.4.43) as well as your [online version](http://qualweb.di.fc.ul.pt/evaluator/):
- https://ast.dk/nyheder/nyheder/nyheder-2020-1/ankestyrelsens-kursuskatalog-for-2021-er-pa-gaden
- https://ast.dk/naevn/adoptionsnaevnet/afgorelser-adoptionsnaevnet
- https://ast.dk/naevn/adoptionsnaevnet/udgivelser-fra-adoptionsnaevnet/adoptionsnaevnets-landsmodereferater
- https://ast.dk/filer/naevn/adoptionsnaevnet/adoptionsnaevnets-landsmodereferater/landsmode-2006.pdf/view
- https://ast.dk/om-ankestyrelsen/artikler-om-styrelsen-generelt-fra-ankestyrelsens-faglig-nyhedsbrev-nyt-fra-ankestyrelsen/vores-opgaver/artikel-2018-i-tal-kommunernes-afgorelser-pa-social-og-beskaeftigelsesomradet
- https://ast.dk/nyheder/nyheder/nyheder-2018/adopterede-og-soskende-i-danmark

Example output from stderr:
```
Error: Evaluation failed: SyntaxError: Invalid regular expression: /^rotate(/: Unterminated group
    at String.match (<anonymous>)
    at String.startsWith (https://ast.dk/++plone++production/++unique++2020-04-06T12:47:58.230305/default.js:2546:34762)
    at QW_ACT_R7.execute (webpack://ACTRules/./dist/rules/QW-ACT-R7.js?:207:27)
    at QW_ACT_R7.ElementIsVisible.descriptor.value (webpack://ACTRules/./dist/lib/decorator.js?:190:27)
    at QW_ACT_R7.ElementExists.descriptor.value (webpack://ACTRules/./dist/lib/decorator.js?:34:27)
    at ACTRules.executeRule (webpack://ACTRules/./dist/index.js?:107:56)
    at ACTRules.executePageMappedRules (webpack://ACTRules/./dist/index.js?:127:40)
    at ACTRules.executeConcurrentRules (webpack://ACTRules/./dist/index.js?:192:14)
    at ACTRules.execute (webpack://ACTRules/./dist/index.js?:218:14)
    at __puppeteer_evaluation_script__:3:28
    at ExecutionContext._evaluateInternal (<source location on disk>\
ode_modules\\\u001b[4mpuppeteer\u001b[24m\\lib\\cjs\\puppeteer\\common\\ExecutionContext.js:217:19)
\u001b[90m    at processTicksAndRejections (node:internal/process/task_queues:93:5)\u001b[39m
    at ExecutionContext.evaluate (<source location on disk>\
ode_modules\\\u001b[4mpuppeteer\u001b[24m\\lib\\cjs\\puppeteer\\common\\ExecutionContext.js:106:16)
```

Site 2
------

The second site seems to run through your online evaluator just fine, but fails in our environments. Sample URLs:

- https://www.sst.dk/da/Viden/Forebyggelse/Forebyggelsespakker-til-kommunerne/Forvaltninger-paa-boerneomraadet/Ungdomsuddannelser
- https://www.sst.dk/da/Viden/Smitsomme-sygdomme/Smitsomme-sygdomme-A-AA/Difteri
- https://www.sst.dk/da/Viden/Specialeplanlaegning/Gaeldende-specialeplan/Specialeplan-for-anaestesiologi
- https://www.sst.dk/da/Viden/Smitsomme%20sygdomme/Smitsomme%20sygdomme%20A-AA/Influenza
- https://www.sst.dk/da/Viden/Miljoe/Miljoe%20og%20sundhed/Luftforurening

From stderr:
```
Error: Evaluation failed: DOMException: Failed to execute 'querySelector' on 'Document': '#4A531B773DA048F096A3E26717A216EE-label' is not a valid selector.
    at QWPage.getElementByID (webpack://QWPage/./dist/index.js?:152:39)
    at verifyAriaLabel (webpack://ACTRules/./node_modules/@qualweb/util/dist/accessibilityUtils/getAccessibleNameRecursion.js?:217:27)
    at Object.getAccessibleNameRecursion [as default] (webpack://ACTRules/./node_modules/@qualweb/util/dist/accessibilityUtils/getAccessibleNameRecursion.js?:17:35)
    at Function.getAccessibleNameRecursion (webpack://ACTRules/./node_modules/@qualweb/util/dist/accessibilityUtils/accessibilityUtils.js?:50:52)
    at Function.descriptor.value (webpack://ACTRules/./node_modules/@qualweb/util/dist/decorator.js?:16:33)
    at Object.getAccessibleName [as default] (webpack://ACTRules/./node_modules/@qualweb/util/dist/accessibilityUtils/getAccessibleName.js?:5:38)
    at Function.getAccessibleName (webpack://ACTRules/./node_modules/@qualweb/util/dist/accessibilityUtils/accessibilityUtils.js?:47:43)
    at Function.descriptor.value (webpack://ACTRules/./node_modules/@qualweb/util/dist/decorator.js?:16:33)
    at Object.getElementRole [as default] (webpack://ACTRules/./node_modules/@qualweb/util/dist/accessibilityUtils/getElementRole.js?:5:43)
    at Function.getElementRole (webpack://ACTRules/./node_modules/@qualweb/util/dist/accessibilityUtils/accessibilityUtils.js?:56:40)
    at ExecutionContext._evaluateInternal (<source location on disk>/node_modules/\u001b[4mpuppeteer\u001b[24m/lib/cjs/puppeteer/common/ExecutionContext.js:217:19)
    at runMicrotasks (<anonymous>)
\u001b[90m    at processTicksAndRejections (internal/process/task_queues.js:93:5)\u001b[39m
    at ExecutionContext.evaluate (<source location on disk>/node_modules/\u001b[4mpuppeteer\u001b[24m/lib/cjs/puppeteer/common/ExecutionContext.js:106:16)
    at Evaluation.executeACT (<source location on disk>/node_modules/\u001b[4m@qualweb\u001b[24m/evaluation/dist/index.js:88:27)
    at Evaluation.evaluatePage (<source location on disk>/node_modules/\u001b[4m@qualweb\u001b[24m/evaluation/dist/index.js:163:57)
```

We've seen this error sporadically on other sites, but none quite as severely as on this one.

---

Do let me know if I can offer more information!

**Comment by carlosapaduarte** [comment]
Hey @Tellus 
Can you update to the latest version of qualweb/core and tell us if the problems persist?

**Comment by Tellus** [comment]
Apologies for the late response

The recurring problems for those sites were indeed resolved with version 0.4.46

Thanks again for the great work :)
